### PR TITLE
Adds saver/loader for excel in pandas extensions (#342)

### LINKED ISSUE #342: pandas excel
1. Implement https://pandas.pydata.org/docs/reference/io.html#excel as a data saver/loader.
It should go into pandas_extensions.py, one class for the writer, on class for the reader, e.g. much like how the [pandas pickle reader and writer](https://github.com/DAGWorks-Inc/hamilton/blob/main/hamilton/plugins/pandas_extensions.py) are structured.
2. There should be requisite tests to exercise the functionality.
3. Then an example should be added to the [example/pandas/materialization example](https://github.com/DAGWorks-Inc/hamilton/tree/main/examples/pandas/materialization). It should just add to the existing materializers; yes this only exercises the writer.

### PR DESCRIPTION
**Pull request by tyapochkin**
This PR adds Pandas data saver and data loader for Excel format and closes #342.

## Changes

- `hamilton/plugins/pandas_extensions.py`: added PandasExcelWriter and PandasExcelReader classes
- `tests/plugins/test_pandas_extensions.py`: added new tests for these classes
- `examples/pandas/materialization/my_script.py`: updated the example with Excel materializers
- `examples/pandas/materialization/notebook.ipynb`: updated the example with Excel materializers
- `requirements-test.txt`: added openpyxl
- `tests/resources/data/test_load_from_data.xlsx`: example Excel file for tests

## How I tested this
I added and ran standard tests similar to tests for other data loaders.

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Comment by tyapochkin** [comment]
I see some problems with Python 3.8. I'll fix them.

File: hamilton/plugins/pandas_extensions.py [hunk 1]
```diff
@@ -1409,6 +1409,119 @@ def name(cls) -> str:
...
+    engine_kwargs: Optional[Dict[str, Any]] = None
+
+    @classmethod
+    def applicable_types(cls) -> Collection[Type]:
+        # Returns type for which data loader is available
+        return [DATAFRAME_TYPE]
+
+    def _get_loading_kwargs(self) -> Dict[str, Any]:
+        # Puts kwargs in a dict
+        kwargs = dataclasses.asdict(self)
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:1457]: nice!

**Comment by tyapochkin** [comment]
> I see some problems with Python 3.8. I'll fix them.

Done.

**Review by skrawcz — APPROVED** [review]

**Comment by skrawcz** [comment]
Thanks @tyapochkin ! This should go out next week.

**Pull request merged**